ethportal-peertest/src/main.rs

```diff
@@ -78,27 +134,14 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
         .await;
 
         tokio::spawn(events.process_discv5_requests());
-
-        let target_node: Enr = peertest_config.target_node.parse().unwrap();
-
-        // Test history and state network Ping request on target node
-        info!("Pinging {} on history network", target_node);
-        let ping_result = history_overlay.send_ping(target_node.clone(), None).await;
-        match ping_result {
-            Ok(val) => info!("Successful ping to History network: {:?}", val),
-            Err(msg) => panic!("Invalid ping to History network: {:?}", msg),
-        }
-
-        info!("Pinging {} on state network", target_node);
-        let ping_result = state_overlay.send_ping(target_node.clone(), None).await;
-        match ping_result {
-            Ok(val) => info!("Successful ping to State network: {:?}", val),
-            Err(msg) => panic!("Invalid ping to State network: {:?}", msg),
-        }
-
+        tokio::spawn(async { history_network_task.unwrap().await });
+        tokio::spawn(async { state_network_task.unwrap().await });
+        tokio::spawn(async { jsonrpc_server_task.await.unwrap() });
+        // short sleep so server task gets running
+        thread::sleep(time::Duration::from_millis(1000));
```

Comment:
Sleeping to wait for something to happen will eventually cause heartache: it adds unnecessary slowdown time on launch at times, or even worse causes some funny race condition when the server takes longer than 1s to launch.

Ideally we want something that we can await, which returns when the server is launched. Maybe that means a one-shot channel that sends a signal when the server is up?

My approach would be to just try it out with a short timebox. See if you can bang it out in an hour, without too much code. If not, then push it into an issue (now explaining what you learned from a quick attempt).